### PR TITLE
Mqc galaxy addition

### PR DESCRIPTION
**Pull request by StevenVerbruggen**

### CODE REVIEW

File: tools/rna_tools/mqc/mQC.xml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+<tool id="mQC" name="mQC" version="1.2">
+    <description>quality control of ribosome profiling mapping results</description>
+    <requirements>
+        <requirement type="set_environment">SCRIPT_PATH</requirement>
```
Comment (bgruening) [inline tools/rna_tools/mqc/mQC.xml]: script path is not needed anymore, I guess

File: tools/rna_tools/mqc/mQC.xml [hunk 2]
```diff
@@ -0,0 +1,109 @@
+<tool id="mQC" name="mQC" version="1.2">
```
Comment (bgruening) [inline tools/rna_tools/mqc/mQC.xml]: ids should be lower case

File: tools/rna_tools/mqc/mQC.xml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    <description>quality control of ribosome profiling mapping results</description>
+    <requirements>
+        <requirement type="set_environment">SCRIPT_PATH</requirement>
+        <requirement type="package" version="1.3.1">samtools</requirement>
+        <requirement type="package" version="1.2">mqc</requirement>
+    </requirements>
+    <command interpreter="perl">
+        mQC.pl --galaxy Y --samfile $samfile --experiment_name $exp_name --cores $cores --species $species --ens_v $ensv --unique $uniq --mapper $mapper --maxmultimap $maxmultimap --ens_db "get" --offset $offsetSelect.offset --min_length_gd $min_l_gd --max_length_gd $max_l_gd --tool_dir "/srv/galaxy/server/tool-data/mQC/mqc_tools/" --plotrpftool $plotrpftool --outhtml $MappingQC_output --outfolder $MappingQC_output.files_path
```
Comment (bgruening) [inline tools/rna_tools/mqc/mQC.xml]: all input, output and type="text" should be single quoted `'`

File: tools/rna_tools/mqc/mQC.xml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        
+        --plastid_bam $offsetSelect.plastid_bam
+        --min_length_plastid $offsetSelect.min_l_plastid
+        --max_length_plastid $offsetSelect.max_l_plastid
+        
+        #end if
+        
+    </command>
+    <inputs>
+        <param name="samfile" type="data" format="sam" label="Choose the sam file on which you want to do quality control"/>
```
Comment (bgruening) [inline tools/rna_tools/mqc/mQC.xml]: Does this also work on BAM files?

File: tools/rna_tools/mqc/mQC.xml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        <param name="exp_name" size="20" type="text" label="Enter the experiment name"/>
```
Comment (bgruening) [inline tools/rna_tools/mqc/mQC.xml]: size="" is deprecated and not used anymore you can remove it.

File: tools/rna_tools/mqc/mQC.xml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        <param name="ensv" type="integer" size="5" value="86" label="Enter the Ensembl database version"/>
+        <param name="uniq" type="select" label="Perform mQC with unique mapped reads only">
+            <option value="Y" selected="true">Yes</option>
+            <option value="N">No</option>
+        </param>
+        <param name="plotrpftool" type="select" label="Choose the style of the RPF phase plot">
+            <option value="grouped2D" select="true">Grouped 2D bar chart</option>
+            <option value="pyplot3D">3D bar chart (may suffer from Escher effects)</option>
+        </param>
+        <param name="mapper" type="select" label="Which mapper is used to generate your SAM file?">
```
Comment (bgruening) [inline tools/rna_tools/mqc/mQC.xml]: `was used`?
Is HiSAT2 also supported?

File: tools/rna_tools/mqc/mQC.xml [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            <param name="offset" type="select" label="The way P-site offsets were determined">
+                <option value="standard" selected="true">Standard offsets</option>
+                <option value="plastid">Calculate offsets with plastid</option>
+                <option value="from_file">Offsets from custom file</option>
+            </param>
+            <when value="from_file">
+                <param name="offsetFile" type="data" format="tabular" label="Choose a text file with the offsets. Each line should be structured as RPF length - offset, seperated by a tab or a space" />
+            </when>
+            <when value="plastid">
+                <param name="plastid_bam" type="data" format="bam" label="Select the mapping bam file for offset calculation" />
```
Comment (bgruening) [inline tools/rna_tools/mqc/mQC.xml]: Here the input is only BAM, what about SAM?

File: tools/rna_tools/mqc/mQC.xml [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+------
+
+.. class:: infomark
+
+**Citation**
+
+For the underlying tool and its use in Galaxy, cite:
+::
+
+    Verbruggen, Steven and Menschaert, Gerben. 2017. "MappingQC: a tool for quality control of ribosome profiling mapping results". Unpublished.
```
Comment (bgruening) [inline tools/rna_tools/mqc/mQC.xml]: Can you use the `citations` tag for this. Just make up one with bibtex.

File: tools/rna_tools/mqc/mQC.xml [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+                <param name="plastid_bam" type="data" format="bam" label="Select the mapping bam file for offset calculation" />
+                <param name="min_l_plastid" value="22" type="integer" label="Enter the minimum RPF length for plastid offset calculation"/>
+                <param name="max_l_plastid" value="34" type="integer" label="Enter the maximum RPF length for plastid offset calculation"/>
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data format="html" name="MappingQC_output" label="HTML output" />
+    </outputs>
+    <help>
```
Comment (bgruening) [inline tools/rna_tools/mqc/mQC.xml]: Tests are missing in your tools. 
```
<tests>
<test>
</test>
</tests>
```

File: tools/rna_tools/mqc/LICENSE [hunk 10]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (bgruening) [inline tools/rna_tools/mqc/LICENSE]: his can be removed

File: tools/rna_tools/mqc/mQC.xml (hunk 4 above)
Comment (StevenVerbruggen) [inline tools/rna_tools/mqc/mQC.xml]: I could include an option that you can input a BAM file and that mQC uses samtools to convert this input BAM file to a SAM file to work on. Is this desirable?
An what do I put then in the format="sam" option of the XML file?

**Comment by bgruening** [comment]
The error looks like the wrong conda version is used, which is expected as the Conda package is not yet merged, so far so good :)

File: tools/rna_tools/mqc/mQC.xml [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+        --species $species
+        --ens_v $ensv
+        --unique $uniq
+        --mapper $mapper
+        --maxmultimap $maxmultimap
+        --ens_db "get"
+        --offset $offsetSelect.offset
+        --min_length_gd $min_l_gd
+        --max_length_gd $max_l_gd
+        ##--tool_dir "/srv/galaxy/server/tool-data/mQC/mqc_tools/"
```
Comment (bgruening) [inline tools/rna_tools/mqc/mQC.xml]: @StevenVerbruggen is this option needed?

**Comment by bgruening** [comment]
Interesting we got no output in 10min. How long is this tool supposed to run?

**Comment by StevenVerbruggen** [comment]
The tool runs a bit longer than one hour if you work with standard offsets (galaxy test runs with standard offsets). We tried to include a 'fast run' option, working with a subset of the samfile , but as you need to construct the phasing over all canonical transcripts out of the annotation anyway and as this takes the bigger part of the running time, it was useless to include this fast run option.
If you want to calculate plastid offsets, 2 hours extra come in to do so.

**Comment by bgruening** [comment]
@StevenVerbruggen `JobOutputsError: [Errno 21] Is a directory: '/home/travis/build/bgruening/galaxytools/tools/rna_tools/mqc/test-data/mQC_test_data_mESC'`

**Comment by bgruening** [comment]
This is awesome @StevenVerbruggen! Sorry that it took so long. I think we have no a really nice high quality tool with conda and testing all over. Thanks so much for staying motivated :)

**Pull request merged**